mono/metadata/appdomain.c

```diff
@@ -3010,12 +3010,11 @@ unload_thread_main (void *arg)
 	int i;
 	gsize result = 1; // failure
 
-	internal = mono_thread_internal_current ();
-
-	MonoString *thread_name_str = mono_string_new_checked (mono_domain_get (), "Domain unloader", error);
-	if (is_ok (error))
-		mono_thread_set_name_internal (internal, thread_name_str, TRUE, FALSE, error);
-	if (!is_ok (error)) {
+	// Domain unloader
+	mono_thread_set_name_constant (mono_thread_internal_current (),
+		MonoSetThreadNameFlag_Permanent, error,
+		'D','o','m','a','i','n',' ','u','n','l','o','a','d','e','r');
```

Comment:
Can we just pass a normal c string here ? What does this change achieve ?

Reply:
It gets us utf8 and utf16 static const. No copy, no alloc, no conversion.

Reply:
I don't think is really neccessary, it looks ugly, and it saves like 1 allocation during the whole lifetime of the program.

Reply:
I changed every constant thread name.
For example, `worker_callback` sets the thread name in a loop.
I think that is kinda terrible but it was defended as correct a while ago.

I have a larger PR where we can pass around utf8 and/or utf16, keep whatever we are given, convert to what we aren't, which we kinda do anyway. And support Windows native thread names, which are always unicode. So I'd like to start with this one and build up to that.

I agree it is a little ugly. I'd like to save whatever allocate and copy is easy. This is easy. Albeit ugly.
In C++11 you can say u"foo" to get a ucs2 string on all systems. That might be nice to make this pretty again some day.
On Win32 you could always say L"foo" but unfortunately that is ucs4 on others.

And this is crashing now I have to figure out why.

Reply:
Ok I have another idea to solve a few matters here. Please wait a day.
The strings will look “normal”.
Windows under ifdef will also use L.
Managed will pass utf8 & utf16.
Only utf8 will be retained.
Still a flag to distinguish constants.
Still curious why this crash.

Reply:
Managed still just pass utf16, but the conversion will be moved up to `ves_icall_System_Threading_Thread_SetName_icall` instead of `mono_thread_set_name_internal`. Subtle.
This way native owns the alloc and free.